Sources/NIOSSL/SSLContext.swift

```diff
@@ -173,6 +287,25 @@ public final class NIOSSLContext {
         // Cipher suites. We just pass this straight to BoringSSL.
         returnCode = CNIOBoringSSL_SSL_CTX_set_cipher_list(context, configuration.cipherSuites)
         precondition(1 == returnCode)
+        
+        // Setup PSK support if a `PSKIdentityCallbackManager` is passed in.
+        if let pskCallback = configuration.pskCallback {
+            self.pskIdentityCallbackManager = pskCallback
+            
+            // Try to detect if the client callback is available to set the BoringSSL client callback
+            if let clientSideCallback = pskCallback.clientCallback {
+                let defaultClientPSKInfo = clientSideCallback(nil)
+                let indentityBytes = defaultClientPSKInfo.identity.backing.withUnsafeBytes{  Array($0) }
+                CNIOBoringSSL_SSL_CTX_use_psk_identity_hint(context, String(decoding: indentityBytes, as: Unicode.UTF8.self))
+                CNIOBoringSSL_SSL_CTX_set_psk_client_callback(context, { clientPSKCallback(ssl: $0, hint: $1, identity: $2, max_identity_len: $3, psk: $4, max_psk_len: $5 ) } )
+            // Ohterwise see if the server side callback is present.
+            } else if let serverSideCallback = pskCallback.serverCallback {
+                let defaultServerPSKInfo = serverSideCallback(nil, nil) // nil, nil being passed in as a way to pass back default data.
+                let indentityBytes = defaultServerPSKInfo.identity.backing.withUnsafeBytes{  Array($0) }
+                CNIOBoringSSL_SSL_CTX_use_psk_identity_hint(context, String(decoding: indentityBytes, as: Unicode.UTF8.self))
```

Comment:
This isn't great: we should almost certainly just let the PSK hint be configured on the `TLSConfiguration` directly.

Reply:
Resolved in 409e1c1